content/ko/docs/concepts/configuration/configmap.md

```diff
@@ -0,0 +1,169 @@
+값이 구성 형식의 조각처럼 보이는 다른 키를 가진 컨피그맵의
+예시이다.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  Name: game-demo
+data:
+  # 속성과 비슷한 키; 각 키는 간단한 값으로 매핑된다
+  player_initial_lives: 3
+  ui_properties_file_name: "user-interface.properties"
+  #
+  # 파일과 비슷한 키
+  game.properties: |
+    enemy.types=aliens,monsters
+    player.maximum-lives=5
+  user-interface.properties: |
+    color.good=purple
+    color.bad=yellow
+    allow.textmode=true
+```
+
+컨피그맵을 사용하여 파드 내부에 컨테이너를 구성할 수 있는
+네 가지 방법이 있다.
+
+1. 컨테이너의 엔트리포인트에 대한 커맨드 라인 인수
+1. 컨테이너에 대한 환경 변수
+1. 애플리케이션이 읽을 수 있도록 읽기 전용 볼륨에 파일 추가
+1. 쿠버네티스 API를 사용하여 컨피그맵을 읽는 파드 내에서 실행할 코드 작성
+
+이러한 다른 방법은 소비되는 데이터를 모델링하는
+다른 방법에 적합하다.
+처음 세 가지 방법의 경우,
+{{< glossary_tooltip text="kubelet" term_id="kubelet" >}}은 파드의 컨테이너를 시작할 때
+시크릿의 데이터를 사용한다.
+
+네 번째 방법은 시크릿과 데이터를 읽기 위해 코드를 작성해야 한다는 것을 의미한다.
+그러나, 쿠버네티스 API를 직접 사용하기 때문에, 애플리케이션은
+컨피그맵이 변경될 때마다 업데이트를 받기 위해 구독할 수 있고, 업데이트가
+있으면 반응한다. 쿠버네티스 API에 직접 접근하면, 이
+기술을 사용하여 다른 네임스페이스의 컨피그맵에 접근할 수도 있다.
+
+다음은 `game-demo` 의 값을 사용하여 파드를 구성하는 파드 예시이다.
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: configmap-demo-pod
+spec:
+  containers:
+    - name: demo
+      image: game.example/demo-game
+      env:
+        # 환경 변수 정의
+        - name: PLAYER_INITIAL_LIVES # 참고로 여기서는 컨피그맵의 키 이름과
+                                     # 대소문자가 다르다.
+          valueFrom:
+            configMapKeyRef:
+              name: game-demo           # 이 값의 컨피그맵.
+              key: player_initial_lives # 가져올 키.
+        - name: UI_PROPERTIES_FILE_NAME
+          valueFrom:
+            configMapKeyRef:
+              name: game-demo
+              key: ui_properties_file_name
+      volumeMounts:
+      - name: config
+        mountPath: "/config"
+        readOnly: true
+  volumes:
+    # 파드 레벨에서 볼륨을 설정한 다음, 해당 파드 내의 컨테이너에 마운트한다.
+    - name: config
+      configMap:
+        # 마운트하려는 컨피그맵의 이름을 제공한다.
+        name: game-demo
+```
+
+
+컨피그맵은 단일 라인 속성 값과 멀티 라인의 파일과 비슷한 값을
+구분하지 않는다.
+파드와 다른 오브젝트가 이러한 값을 소비하는 방식이 중요하다.
+이 예제에서, 볼륨을 정의하고 `demo` 컨테이너에
+`/config` 로 마운트하면 4개의 파일이 생성된다.
+
+- `/config/player_initial_lives`
+- `/config/ui_properties_file_name`
+- `/config/game.properties`
+- `/config/user-interface.properties`
+
+`/config` 에 `.properties` 확장자를 가진 파일만
+포함시키려면, 두 개의 다른 컨피그맵을 사용하고, 파드에
+대해서는 `spec` 의 두 컨피그맵을 참조한다. 첫 번째 컨피그맵은
+`player_initial_lives` 와 `ui_properties_file_name` 을 정의한다. 두 번째
+컨피그맵은 kubelet이 `/config` 에 넣는 파일을 정의한다.
+
+{{< note >}}
+컨피그맵을 사용하는 가장 일반적인 방법은 동일한 네임스페이스의
+파드에서 실행되는 컨테이너에 대한 설정을 구성하는 것이다. 컨피그맵을
+별도로 사용할 수도 있다.
+
+예를 들어,
+컨피그맵에 기반한 동작을 조정하는 {{< glossary_tooltip text="애드온" term_id="addons" >}}이나
+{{< glossary_tooltip text="오퍼레이터" term_id="operator-pattern" >}}를
+사용할 수도 있다.
+{{< /note >}}
+
+
+{{% /capture %}}
+{{% capture whatsnext %}}
+
+* [시크릿](/docs/concepts/configuration/secret/)에 대해 읽어본다.
+* [컨피그맵을 사용하도록 파드 구성하기](/docs/tasks/configure-pod-container/configure-pod-configmap/)를 읽어본다.
+* 코드를 구성에서 분리하는 동기를 이해하려면
+  [Twelve-Factor 앱](https://12factor.net/)을 읽어본다.
```

Comment:
Optionally, hyperlink to https://12factor.net/ko/

Reply:
@sftim , Thank you for your review.